### PR TITLE
Optimized prerendered version of the TextField

### PR DESCRIPTION
**Pull request by cromefire**
The prerendered version is now using placeholder instead off an additional label

### CODE REVIEW

**Comment by prateekbh** [comment]
@cromefire is this ready? do you want this in 1.5.3? or in 2.0?

**Comment by cromefire** [comment]
Let's do last update, maybe fix some bugs and that work towards `2.0`

**Comment by prateekbh** [comment]
so u want this in 1.5.3, right?

**Comment by cromefire** [comment]
yes

File: ts/TextField/index.tsx [hunk 1]
```diff
@@ -284,10 +286,6 @@ export class TextField extends Component<
 
     return showDiv ? (
       <div className={className} style={outerStyle}>
-        {props.label &&
```
[the remark this answers is not in the document]
Comment (prateekbh) [reply ts/TextField/index.tsx]: gist: if no `jsComponentAttached` then label should be outside the container div of the input

[the remark this answers is not in the document]
Comment (cromefire) [reply ts/TextField/index.tsx]: > gist: if no jsComponentAttached then label should be outside the container div of the input

The problem is: When the page is rehydrated the whole layout changes (and the visual difference is huge).

Yes only a placeholder is not a good user experience but it is not expected that the user is fast enough to edit all the things, when the page isn't loaded yet. It's temporary until the page has loaded completely.

> Y remove this? Isnt it how it should be if jS is not present?

So yes, but **only if it is not expected to be used with js**. This is temporary during load, there is currently no way to use this component without the js component in pmc.

**Pull request merged**